model/project_parser.go

```diff
@@ -1311,32 +1326,39 @@ func evaluateBVTasks(tse *taskSelectorEvaluator, tgse *tagSelectorEvaluator, vse
 		// only error if both selectors error because each task should only be found
 		// in one or the other.  Skip selector checking if task group is defined
 		// directly on the build variant task.
-		var names, temp []string
+		var names, unmatched, temp []string
 		isGroup := false
 		if pbvt.TaskGroup != nil {
 			names = append(names, pbvt.Name)
 			isGroup = true
 		} else {
 			var err1, err2 error
 			if tse != nil {
-				temp, err1 = tse.evalSelector(ParseSelector(pbvt.Name))
+				temp, unmatched, err1 = tse.evalSelector(ParseSelector(pbvt.Name))
 				names = append(names, temp...)
 			}
 			if tgse != nil {
-				temp, err2 = tgse.evalSelector(ParseSelector(pbvt.Name))
+				var unmatchedTaskgroup []string
+				temp, unmatchedTaskgroup, err2 = tgse.evalSelector(ParseSelector(pbvt.Name))
+				unmatched = append(unmatched, unmatchedTaskgroup...)
 				if len(temp) > 0 {
 					names = append(names, temp...)
 					isGroup = true
 				}
 			}
 			if err1 != nil && err2 != nil {
 				evalErrs = append(evalErrs, err1, err2)
-				emptySelectors = append(emptySelectors, pbvt.Name)
+				unmatchedSelectorLines = append(unmatchedSelectorLines, pbvt.Name)
 				continue
 			}
 		}
 		if len(names) == 0 {
-			emptySelectors = append(emptySelectors, pbvt.Name)
+			unmatchedSelectorLines = append(unmatchedSelectorLines, pbvt.Name)
+		}
+		for _, tag := range unmatched {
```

Comment:
Does this check do something extra for the unmatched criteria? It seems like the `evalSelector`s above this already check for matching literal task/task group name.

Reply:
If it's matched via task or task group, it still will append it to unmatched because of the other one that it doesn't match to- so we have to make sure it wasn't matched by either by checking if the currently appended found tasks/task groups (`names`) has it